lang/src/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransform.kt

```diff
@@ -0,0 +1,167 @@
+                                    is PartiqlAst.ProjectItem.ProjectExpr ->
+                                        structField(
+                                            lit(
+                                                projectItem.asAlias?.toIonElement()
+                                                    ?: errAstNotNormalized("SELECT-list item alias not specified")
+                                            ),
+                                            transformExpr(projectItem.expr),
+                                        )
+                                    is PartiqlAst.ProjectItem.ProjectAll -> {
+                                        structFields(transformExpr(projectItem.expr), projectItem.metas)
+                                    }
+                                }
+                            }
+                        )
+                    }
+                    is PartiqlAst.Projection.ProjectStar ->
+                        // `SELECT * FROM bar AS b` is rewritten to `SELECT b.* FROM bar as b` by
+                        // [SelectStarVisitorTransform]. Therefore, there is no need to support `SELECT *` here.
+                        errAstNotNormalized("Expected SELECT * to be removed")
+
+                    is PartiqlAst.Projection.ProjectPivot -> TODO("PIVOT ...")
+                },
+                algebra,
+                node.project.metas
+            )
+        }.let { q ->
+            // in case of SELECT DISTINCT, wrap bindingsToValues in call to filter_distinct
+            when (node.setq) {
+                null, is PartiqlAst.SetQuantifier.All -> q
+                is PartiqlAst.SetQuantifier.Distinct -> PartiqlLogical.build { call("filter_distinct", q) }
+            }
+        }
+
+    /**
+     * Throws [NotImplementedError] if any `SELECT` clauses were used that are not mappable to [PartiqlLogical].
+     *
+     * This function is temporary and will be removed when all the clauses of the `SELECT` expression are mappable
+     * to [PartiqlLogical].
+     */
+    private fun checkForUnsupportedSelectClauses(node: PartiqlAst.Expr.Select) {
+        when {
+            node.group != null -> TODO("Support for GROUP BY")
+            node.order != null -> TODO("Support for ORDER BY")
+            node.having != null -> TODO("Support for HAVING")
+        }
+    }
+
+    override fun transformLetBinding(node: PartiqlAst.LetBinding): PartiqlLogical.LetBinding =
+        PartiqlLogical.build {
+            letBinding(
+                transformExpr(node.expr),
+                varDecl_(node.name, node.name.metas),
+                node.metas
+            )
+        }
+
+    override fun transformStatementDml(node: PartiqlAst.Statement.Dml): PartiqlLogical.Statement {
+        TODO("Support for DML")
+    }
+
+    override fun transformStatementDdl(node: PartiqlAst.Statement.Ddl): PartiqlLogical.Statement {
+        TODO("Support for DDL")
+    }
```

Comment:
I'm not really sure about this, but I think DML & DDL are not parts of logical plans. So is it better to throw error for these methods, instead of `TODO`?

Reply:
- DML is definitely needed since is really just another type of query--the results of which are applied to the data store instead of returned to the caller.  That said, it is out of scope for now.
- DDL isn't needed in the short term but there's no reason why it can't be part of the logical and physical plans at some point.  It's also entirely possible that DDL never has a representation in the logical or physical plans.  There may not be a benefit.  I am undecided because I don't have any customer requirements to help guide the decision.